packages/rich-text/src/plugins/List/transforms/unwrapList.ts

```diff
@@ -1,36 +1,97 @@
 /**
  * Credit: Modified version of Plate's list plugin
- * See: https://github.com/udecode/plate/blob/main/packages/nodes/list
+ * See: https://github.com/udecode/plate/blob/main/packages/list/src/transforms/unwrapList.ts
  */
+
 import { BLOCKS } from '@contentful/rich-text-types';
+import { getListTypes } from '@udecode/plate-list';
 
-import { withoutNormalizing } from '../../../internal';
-import { getNodeEntries, isElement } from '../../../internal/queries';
-import { unwrapNodes, liftNodes } from '../../../internal/transforms';
-import { PlateEditor, Path } from '../../../internal/types';
+import {
+  getAboveNode,
+  getBlockAbove,
+  getCommonNode,
+  getPluginType,
+  isElement,
+} from '../../../internal/queries';
+import { withoutNormalizing, unwrapNodes, setElements } from '../../../internal/transforms';
+import { PlateEditor, Path, Descendant } from '../../../internal/types';
 
-function hasUnliftedListItems(editor: PlateEditor, at?: Path) {
-  return getNodeEntries(editor, {
-    at,
-    match: (node, path) => isElement(node) && node.type === BLOCKS.LIST_ITEM && path.length >= 2,
-  }).next().done;
+function handleEmbeddedType(
+  editor: PlateEditor,
+  blockType: BLOCKS.EMBEDDED_ASSET | BLOCKS.EMBEDDED_ENTRY,
+  at?: Path,
+  firstChild?: Descendant
+) {
+  if (firstChild) {
+    setElements(editor, {
+      at,
+      type: getPluginType(editor, blockType),
+      value: { ...firstChild },
+    });
+  }
 }
+
 export const unwrapList = (editor: PlateEditor, { at }: { at?: Path } = {}) => {
+  const ancestorListTypeCheck = () => {
+    if (getAboveNode(editor, { match: { at, type: getListTypes(editor) } })) {
+      return true;
+    }
+    // The selection's common node might be a list type
+    if (!at && editor.selection) {
+      const commonNode = getCommonNode(
+        editor,
+        editor.selection.anchor.path,
+        editor.selection.focus.path
+      );
+
+      if (isElement(commonNode[0]) && getListTypes(editor).includes(commonNode[0].type)) {
+        return true;
+      }
+    }
+
+    return false;
+  };
+
   withoutNormalizing(editor, () => {
     do {
-      // lift list items to the root level
-      liftNodes(editor, {
+      const licEntry = getBlockAbove(editor, {
         at,
-        match: (node) => isElement(node) && node.type === BLOCKS.LIST_ITEM,
-        mode: 'lowest',
+        match: { type: getPluginType(editor, BLOCKS.LIST_ITEM) },
       });
-    } while (!hasUnliftedListItems(editor, at));
 
-    // finally unwrap all lifted items
-    unwrapNodes(editor, {
-      at,
-      match: { type: BLOCKS.LIST_ITEM },
-      split: false,
-    });
+      if (licEntry) {
+        // Special case for embedded entry and asset
```

Comment:
Special handling for embedded assets and entries, aside from this everything else is pretty much the same as the source